llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -5753,7 +5753,8 @@ TargetLowering::ParseConstraints(const DataLayout &DL,
       assert(!Call.getType()->isVoidTy() && "Bad inline asm!");
       if (auto *STy = dyn_cast<StructType>(Call.getType())) {
         OpInfo.ConstraintVT =
-            getSimpleValueType(DL, STy->getElementType(ResNo));
+            getAsmOperandValueType(DL, STy->getElementType(ResNo))
+                .getSimpleVT();
```

Comment:
What happens for nested structs? Are those supported?

Reply:
I don't believe the lowering from an inline assembly statement to a `call ... asm` will create a nested struct. It either returns the single type corresponding to the output (if there's one output), or a struct with a member per output (if there's multiple outputs). I think that inline asm inputs/outputs cannot themselves be structs, but I'm not 100% certain of that. This code would suggest not.

Reply:
I just tried it and it seems to hit unreachable "Unknown type!"; we should do better here and report a proper error

Reply:
Better errors for IR that frontends don't produce seems like a different issue to fixing an actual crash in the AArch64 backend that frontends can hit, so should be in a different PR IMO.

Reply:
Yes, separate fix